In [1]:
import requests
import pandas as pd
import numpy as np
import time 
import json

In [2]:
BTC = requests.get("https://api.cryptowat.ch/markets/kraken/btcusd/ohlc?periods=86400&apikey=XVGSK3GTQSR4WA9W8XY7")
LTC = requests.get("https://api.cryptowat.ch/markets/kraken/ltcusd/ohlc?periods=86400&apikey=XVGSK3GTQSR4WA9W8XY7")
ETH = requests.get("https://api.cryptowat.ch/markets/kraken/ethusd/ohlc?periods=86400&apikey=XVGSK3GTQSR4WA9W8XY7") 
XRP = requests.get("https://api.cryptowat.ch/markets/kraken/xrpusd/ohlc?periods=86400&apikey=XVGSK3GTQSR4WA9W8XY7") 
DOGE = requests.get("https://api.cryptowat.ch/markets/kraken/dogeusd/ohlc?periods=86400&apikey=XVGSK3GTQSR4WA9W8XY7") 

In [3]:
def clean_api_request(req):
    req = req.json()
    # Convert jsonized lists to DF 
    req = pd.DataFrame(req['result']['86400'])
    # Rename DF columns 
    req.columns = ['date', 'open_price', 'hi_price', 'lo_price', 'close_price', 'vol', 'quote_vol']
    # Select only date and closing value 
    req = req[['date','close_price']]
    # Convert Epoch time (secs since 1970) into datetime object
    req['date'] = pd.to_datetime(req['date'], unit='s')
    #Select date range from 2016 - 2021
    req = (req.loc[req["date"].between("2016-01-01", "2021-12-31")]).reset_index(drop=True)
    return req

In [4]:
display("BITCOIN", clean_api_request(BTC))
display("LITECOIN", clean_api_request(LTC))
display("ETHEREUM", clean_api_request(ETH))
display("RIPPLE XRP", clean_api_request(XRP))
display("DOGECOIN", clean_api_request(DOGE))

'BITCOIN'

date  close_price
0    2016-01-01    431.31000
1    2016-01-02    431.50530
2    2016-01-03    434.98000
3    2016-01-04    426.57000
4    2016-01-05    431.08414
...         ...          ...
2186 2021-12-27  50797.30000
2187 2021-12-28  50687.40000
2188 2021-12-29  47548.10000
2189 2021-12-30  46473.20000
2190 2021-12-31  47143.80000

[2191 rows x 2 columns]

'LITECOIN'

date  close_price
0    2016-01-01      3.48010
1    2016-01-02      3.48500
2    2016-01-03      3.50000
3    2016-01-04      3.49300
4    2016-01-05      3.43589
...         ...          ...
2186 2021-12-27    156.00000
2187 2021-12-28    155.84000
2188 2021-12-29    145.64000
2189 2021-12-30    145.48000
2190 2021-12-31    148.06000

[2191 rows x 2 columns]

'ETHEREUM'

date  close_price
0    2016-01-01      0.94885
1    2016-01-02      0.92037
2    2016-01-03      0.95000
3    2016-01-04      0.96000
4    2016-01-05      0.00000
...         ...          ...
2186 2021-12-27   4064.33000
2187 2021-12-28   4038.76000
2188 2021-12-29   3792.09000
2189 2021-12-30   3628.29000
2190 2021-12-31   3710.00000

[2191 rows x 2 columns]

'RIPPLE XRP'

date  close_price
0    2017-05-19     0.372797
1    2017-05-20     0.320106
2    2017-05-21     0.344006
3    2017-05-22     0.323489
4    2017-05-23     0.303060
...         ...          ...
1682 2021-12-27     0.920180
1683 2021-12-28     0.927700
1684 2021-12-29     0.851790
1685 2021-12-30     0.817040
1686 2021-12-31     0.838770

[1687 rows x 2 columns]

'DOGECOIN'

date  close_price
0   2019-12-20     0.002500
1   2019-12-21     0.002020
2   2019-12-22     0.002100
3   2019-12-23     0.002085
4   2019-12-24     0.002232
..         ...          ...
738 2021-12-27     0.190018
739 2021-12-28     0.187814
740 2021-12-29     0.173745
741 2021-12-30     0.167974
742 2021-12-31     0.171183

[743 rows x 2 columns]

XRP and Doge are missing some historiucal data when pulling the API data, there are no records there before 2017, may be becayse the value for those crypto  was too low.

XRP had only 1687 rows (days of trade, so missing before 2017ish) 

DOGE has onlky 743 rows 

I found historical for those 2 cryptos in Kaggle. 
https://www.kaggle.com/datasets/sudalairajkumar/cryptocurrencypricehistory?resource=download&select=coin_XRP.csv

In [5]:
BTC = clean_api_request(BTC)
LTC = clean_api_request(LTC)
ETH = clean_api_request(ETH)
#XRP and DPOGE data pulled from Kaggle 
XRP = pd.read_csv("../data/XRP_p_16-21.csv")
DOGE = pd.read_csv("../data/dogecoin_p_16_21.csv")

In [6]:
BTC.to_csv("../final_data/btc.csv")
LTC.to_csv("../final_data/ltc.csv")
ETH.to_csv("../final_data/eth.csv")
XRP.to_csv("../final_data/xrp.csv")
DOGE.to_csv("../final_data/doge.csv")

In [ ]:
!git add . 

In [ ]:
!git commit -m'in progress'

In [ ]:
!git push